cmd/chaincmp/chaincmp.go

```diff
@@ -223,44 +223,54 @@ func findLastCommonHeight(interrupt <-chan struct{}, clients []*client.Client, s
 	return r, nil
 }
 
-type nodeHeader struct {
-	id     int
-	header *client.Headers
-	err    error
+type nodeBlockInfo struct {
+	id        int
+	blockID   proto.BlockID
+	height    uint64
+	generator proto.Address
+	blockTime uint64
+	err       error
 }
 
 func differentIdsCount(clients []*client.Client, height int) (int, error) {
-	ch := make(chan nodeHeader, len(clients))
-	info := make(map[int]*client.Headers)
+	ch := make(chan nodeBlockInfo, len(clients))
+	info := make(map[int]nodeBlockInfo)
 	m := make(map[proto.BlockID]bool)
 	for i, c := range clients {
 		go func(id int, cl *client.Client) {
 			ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
 			defer cancel()
 			header, resp, err := cl.Blocks.HeadersAt(ctx, uint64(height))
 			if err != nil {
-				ch <- nodeHeader{id, header, err}
-				return
-			}
-			if rc := resp.StatusCode; rc != 200 {
-				ch <- nodeHeader{id, header, errors.Errorf("unexpected response code %d", rc)}
+				if resp != nil && resp.StatusCode == http.StatusNotFound {
+					ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
```

Comment:
Why did you choose exactly 30 sec timeout? Should we maybe write and use a constant for using this timeout here and somewhere else?